### PR TITLE
[android] [application] migrate to sweet API

### LINKED ISSUE #21110: [expo-application] `Application.androidId` constant do not comply with the privacy policies of app markets such as Huawei and Xiaomi.
### Summary

The same issue exists in `react-native-device-info`, but they have fixed it for now, and the details can be found in https://github.com/react-native-device-info/react-native-device-info/issues/1427.

### What platform(s) does this occur on?

Android

### Environment

System:
      OS: macOS 13.0.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
      Yarn: 1.22.18 - ~/.nvm/versions/node/v16.14.2/bin/yarn
      npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
      Watchman: 2022.10.24.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.1, iOS 16.1, macOS 13.0, tvOS 16.1, watchOS 9.1
      Android SDK:
        API Levels: 27, 29, 30, 31
        Build Tools: 27.0.3, 28.0.3, 29.0.2, 30.0.2, 30.0.3, 31.0.0
        System Images: android-29 | Google APIs Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom, android-33 | Google APIs Intel x86 Atom_64
        Android NDK: 22.1.7171670
    IDEs:
      Android Studio: 2021.3 AI-213.7172.25.2113.9123335
      Xcode: 14.1/14B47b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~47.0.8 => 47.0.8 
      react: 18.1.0 => 18.1.0 
      react-native: 0.70.5 => 0.70.5 
    npmGlobalPackages:
      eas-cli: 3.5.2
      expo-cli: 6.0.8
    Expo Workflow: managed

### Minimal reproducible example

none

### PR DESCRIPTION
**Pull request by fobos531**
# Why

continuation of migration of expo modules

**Note**: This PR introduces a breaking change on Android side, in that the `androidId` is no longer a constant, and rather a method - this aims to resolve https://github.com/expo/expo/issues/21110

# How

followed the usual migration steps

# Test Plan

Verified that all tests still pass
CleanShot 2023-05-21 at 16 28 54

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by fobos531** [comment]
Hey @alanjhughes @lukmccall 

Can you let me know whether there's anything else I need to do have this looked at?

File: android/expoview/src/main/java/versioned/host/exp/exponent/ExperiencePackagePicker.kt [hunk 1]
```diff
@@ -119,6 +117,7 @@ object ExperiencePackagePicker : ModulesProvider {
   }
 
   override fun getModulesList(): List<Class<out Module>> = listOf(
+    ApplicationModule::class.java,
```
Comment (lukmccall) [inline android/expoview/src/main/java/versioned/host/exp/exponent/ExperiencePackagePicker.kt:120]: I think, you're missing a import for that class

**Comment by lukmccall** [comment]
Can you fix the CI:
image

**Comment by lukmccall** [comment]
Closing in favor of https://github.com/expo/expo/pull/24568.

**Pull request closed without merging**